From b8ff6133df30418f0c841185960c6369155f9874 Mon Sep 17 00:00:00 2001
From: Mike RE Mallin <mremallin@gmail.com>
Date: Wed, 4 Dec 2024 15:41:24 -0500
Subject: Trigger neighbour probe when the hold timer starts

(cherry picked from commit afd912bdefc0892297fee789b58eead2c82acde1)
Signed-off-by: Tamer Ahmed <tamerahmed@microsoft.com>

diff --git a/zebra/zebra_dplane.c b/zebra/zebra_dplane.c
index c7b7117d7..fe48f7cb7 100644
--- a/zebra/zebra_dplane.c
+++ b/zebra/zebra_dplane.c
@@ -5745,7 +5745,7 @@ enum zebra_dplane_result dplane_local_neigh_add(const struct interface *ifp,
 					  const struct ipaddr *ip,
 					  const struct ethaddr *mac,
 					  bool set_router, bool set_static,
-					  bool set_inactive)
+					  bool set_inactive, bool trigger_probe)
 {
 	enum zebra_dplane_result result = ZEBRA_DPLANE_REQUEST_FAILURE;
 	uint32_t update_flags = 0;
@@ -5754,9 +5754,12 @@ enum zebra_dplane_result dplane_local_neigh_add(const struct interface *ifp,
 
 	if (set_static)
 		update_flags |= DPLANE_NEIGH_SET_STATIC;
-
-	if (set_inactive) {
+	if (set_inactive)
 		update_flags |= DPLANE_NEIGH_SET_INACTIVE;
+
+	if (trigger_probe) {
+		state = DPLANE_NUD_PROBE;
+	} else if (set_inactive) {
 		state = DPLANE_NUD_STALE;
 	} else {
 		state = DPLANE_NUD_REACHABLE;
diff --git a/zebra/zebra_dplane.h b/zebra/zebra_dplane.h
index 854a0e862..e5adb2198 100644
--- a/zebra/zebra_dplane.h
+++ b/zebra/zebra_dplane.h
@@ -1046,7 +1046,7 @@ enum zebra_dplane_result dplane_local_neigh_add(const struct interface *ifp,
 					  const struct ipaddr *ip,
 					  const struct ethaddr *mac,
 					  bool set_router, bool set_static,
-					  bool set_inactive);
+					  bool set_inactive, bool trigger_probe);
 enum zebra_dplane_result dplane_rem_neigh_delete(const struct interface *ifp,
 					     const struct ipaddr *ip);
 
diff --git a/zebra/zebra_evpn_neigh.c b/zebra/zebra_evpn_neigh.c
index 961b1a5d9..7b8a703d9 100644
--- a/zebra/zebra_evpn_neigh.c
+++ b/zebra/zebra_evpn_neigh.c
@@ -245,6 +245,7 @@ static void zebra_evpn_sync_neigh_dp_install(struct zebra_neigh *n,
 	struct interface *ifp;
 	bool set_static;
 	bool set_router;
+	bool trigger_probe = false;
 
 	zns = zebra_ns_lookup(NS_DEFAULT);
 	ifp = if_lookup_by_index_per_ns(zns, n->ifindex);
@@ -277,7 +278,7 @@ static void zebra_evpn_sync_neigh_dp_install(struct zebra_neigh *n,
 			set_static ? " static" : "",
 			set_inactive ? " inactive" : "");
 	dplane_local_neigh_add(ifp, &n->ip, &n->emac, set_router, set_static,
-			       set_inactive);
+			       set_inactive, trigger_probe);
 }
 
 /*
@@ -451,6 +452,14 @@ static void zebra_evpn_neigh_hold_exp_cb(struct event *t)
 
 static inline void zebra_evpn_neigh_start_hold_timer(struct zebra_neigh *n)
 {
+	struct zebra_ns *zns;
+	struct interface *ifp;
+
+	bool set_router = !!CHECK_FLAG(n->flags, ZEBRA_NEIGH_ROUTER_FLAG);
+	bool set_static = zebra_evpn_neigh_is_static(n);
+	bool set_inactive = !!CHECK_FLAG(n->flags, ZEBRA_NEIGH_LOCAL_INACTIVE);
+	bool trigger_probe = true;
+
 	if (n->hold_timer)
 		return;
 
@@ -459,6 +468,12 @@ static inline void zebra_evpn_neigh_start_hold_timer(struct zebra_neigh *n)
 			   n->zevpn->vni, &n->ip, &n->emac, n->flags);
 	event_add_timer(zrouter.master, zebra_evpn_neigh_hold_exp_cb, n,
 			zmh_info->neigh_hold_time, &n->hold_timer);
+
+	zns = zebra_ns_lookup(NS_DEFAULT);
+	ifp = if_lookup_by_index_per_ns(zns, n->ifindex);
+
+	(void)dplane_local_neigh_add(ifp, &n->ip, &n->emac,
+	                             set_router, set_static, set_inactive, trigger_probe);
 }
 
 static void zebra_evpn_local_neigh_deref_mac(struct zebra_neigh *n,
-- 
2.43.0

